extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -96,6 +105,35 @@ private TransactionConfiguration getTransactionConfiguration(InvocationContext i
         return configuration;
     }
 
+    private Map<Class<?>, Annotation> getAdditionalTransactionalConfiguration(InvocationContext ic) {
```

Comment:
I think this should be cached, as it is slow. Ideally we would pre-compute this at build time. 

Also does this work in native mode? I think it will but an integration test would be helpful.